src/main/java/io/tus/java/client/TusUpload.java

```diff
@@ -58,18 +62,35 @@ public void setSize(long size) {
         this.size = size;
     }
 
+    /**
+     * Returns the file specific fingerprint.
+     * @return Fingerprint as String.
+     */
     public String getFingerprint() {
         return fingerprint;
     }
 
+    /**
+     * Sets a fingerprint for this upload. This fingerprint must be unique and file specific, because it is used
+     * for upload identification.
+     * @param fingerprint String of fingerprint information.
+     */
```

Comment:
Wrote this with my TUS knowledge from the calls.

Couldn't we offer to generate UIDs with Java and propagate them back in case of you don't want to this for yourself ?

Reply:
Comment is accurate in my eyes. But I don´t understand your proposals of UID. Can you explain?